src/service.ts

```diff
@@ -177,6 +221,83 @@ export interface AddDeploymentOptions extends ServiceBindOptions {
   readonly port?: number;
 }
 
+class ImportedService extends Construct implements IService {
+  private readonly _name: string;
+
+  public readonly resourceType = 'service';
+
+  constructor(scope: Construct, id: string, name: string) {
+    super(scope, id);
+    this._name = name;
+  }
+
+  public get name(): string {
+    return this._name;
+  }
+
+  public get apiVersion(): string {
+    return k8s.KubeService.GVK.apiVersion;
+  }
+
+  public get apiGroup(): string {
+    return '';
+  }
+
+  public get kind(): string {
+    return k8s.KubeService.GVK.kind;
+  }
+
+  public get resourceName(): string {
+    return this.name;
+  }
+
+  /**
+   * For imported services, we don't have information about the ports.
+   * If specific port information is needed, it should be passed in configuration.
+   */
+  public get ports(): ServicePort[] {
+    return [];
```

Comment:
With the new proposed API, this will return the user provided ports (if any):

```suggestion
    return [...this._ports];
```

Exactly like the managed service does.